contracts/src/v0.8/dev/KeeperRegistryBase.sol

```diff
@@ -174,71 +203,80 @@ abstract contract KeeperRegistryBase is ConfirmedOwner, ExecutionPrevention, Ree
    * @param paymentModel the payment model of default, Arbitrum, or Optimism
    * @param registryGasOverhead the gas overhead used by registry in performUpkeep
    * @param link address of the LINK Token
-   * @param linkEthFeed address of the LINK/ETH price feed
+   * @param linkNativeFeed address of the LINK/NATIVE price feed
    * @param fastGasFeed address of the Fast Gas price feed
    */
   constructor(
     PaymentModel paymentModel,
     uint256 registryGasOverhead,
     address link,
-    address linkEthFeed,
+    address linkNativeFeed,
     address fastGasFeed
   ) ConfirmedOwner(msg.sender) {
     PAYMENT_MODEL = paymentModel;
     REGISTRY_GAS_OVERHEAD = registryGasOverhead;
     LINK = LinkTokenInterface(link);
-    LINK_ETH_FEED = AggregatorV3Interface(linkEthFeed);
+    LINK_NATIVE_FEED = AggregatorV3Interface(linkNativeFeed);
     FAST_GAS_FEED = AggregatorV3Interface(fastGasFeed);
   }
 
   /**
-   * @dev retrieves feed data for fast gas/eth and link/eth prices. if the feed
-   * data is stale it uses the configured fallback price. Once a price is picked
-   * for gas it takes the min of gas price in the transaction or the fast gas
-   * price in order to reduce costs for the upkeep clients.
+   * @dev retrieves feed data for fast gas/eth. if the feed
+   * data is stale it uses the configured fallback price.
    */
-  function _getFeedData() internal view returns (uint256 gasWei, uint256 linkEth) {
-    uint32 stalenessSeconds = s_storage.stalenessSeconds;
+  function _getFasGasFeedData() internal view returns (uint256 gasWei) {
```

Comment:
might be more gas efficient to pass in `s_config` as an argument, assuming the parent function has already read it into memory

Reply:
Yes my plan is to overall gas golf at the end and hyper optimise performUpkeep path, ensuring all storage is read from  / written to once